implementations/rust/ockam/ockam_identity/src/authenticated_table.rs

```diff
@@ -3,12 +3,11 @@ pub mod lmdb;
 pub mod mem;
 
 use ockam_core::async_trait;
+use ockam_core::{AsyncTryClone, Result};
 
 #[async_trait]
-pub trait Storage {
-    type Error: ockam_core::compat::error::Error + Send + Sync + 'static;
-
-    async fn get(&self, id: &str, key: &str) -> Result<Option<Vec<u8>>, Self::Error>;
-    async fn set(&self, id: &str, key: String, val: Vec<u8>) -> Result<(), Self::Error>;
-    async fn del(&self, id: &str, key: &str) -> Result<(), Self::Error>;
+pub trait Storage: AsyncTryClone + Send + Sync + 'static {
```

Comment:
Nothing in `Storage` seems to require these supertraits, so I recommend removing them.